pkg/controllers/state/cluster.go

```diff
@@ -53,6 +53,11 @@ type Cluster struct {
 	nodeClaimNameToProviderID map[string]string               // node claim name -> provider id
 	daemonSetPods             sync.Map                        // daemonSet -> existing pod
 
+	// podAckTimes keeps track of when Karpenter first tries to schedule the pod.
+	// This is orchestrated internally as opposed to writing to state in the cluster to reduce overall
+	// writes to the APIServer, but also since it only needs to be orchestrated cross-controller internally.
+	podAckTimes map[types.NamespacedName]time.Time // pod namespaced name -> when it was first considered for scheduling
```

Comment:
How much does this increase our memory utilization when there is a large scale-up? Users perform 5000pod scale-ups -- when these size of scale-ups happen is storing this data expensive? I doubt it, but we should check

Reply:
Did a scale up to 3000 and the memory fluctuated to 250 to 350m. Without the changes it's pretty much at the exact same level